src/tasks/DockerRunTaskProvider.ts

```diff
@@ -99,4 +99,11 @@ export class DockerRunTaskProvider extends DockerTaskProvider {
             .withQuotedArg(runOptions.image)
             .withArgs(runOptions.command);
     }
+
+    private getVolumeOptions(volume: DockerContainerVolume, isWindows: boolean): string {

```

Comment:
I'd maybe do this; it's a bit hard to follow the nested ternaries:

```typescript
    private getVolumeOptions(volume: DockerContainerVolume, isWindows: boolean): string {
        if (!volume.permissions) {
            return '';
        } else if (!isWindows) {
            return ':' + volume.permissions;
        } else {
            // The 'z' and 'Z' options aren't supported on Windows containers, normalize to simply ro / rw
            switch (volume.permissions as string) {
                case 'ro,Z':
                case 'ro,z':
                    return ':ro';
                case 'rw,Z':
                case 'rw,z':
                    return ':rw';
                default:
                    return ':' + volume.permissions;
            }
        }
    }
```

Reply:
I guess there are also some merge conflicts due to the fix for #3281.

Reply:
Thanks. I've applied this.